### PR TITLE
545 notify raw

### LINKED ISSUE #545: notify_raw
For some reason we didn't implement this.

### PR DESCRIPTION
**Pull request by lastmjs**
closes #545

### CODE REVIEW

File: examples/notify_raw/test/test.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+const canister2 = createActorCanister2('ryjl3-tyaaa-aaaaa-aaaba-cai', {
+    agentOptions: {
+        host: 'http://127.0.0.1:8000'
+    }
+});
+
+const tests: Test[] = [
+    ...large_wasm_deploy('canister1'),
+    ...large_wasm_deploy('canister2'),
```
Comment (dansteren) [inline examples/notify_raw/test/test.ts:19]: You'll need to use the new `deploy` function when you merge main in.

File: index.ts [hunk 2]
```diff
@@ -122,6 +122,12 @@ type ic = {
     msg_cycles_available128: () => nat;
     msg_cycles_refunded: () => nat64;
     msg_cycles_refunded128: () => nat;
+    notify_raw: (
+        canister_id: Principal,
+        method: string,
+        args_raw: blob,
+        payment: nat
```
Comment (dansteren) [inline index.ts:129]: Is this payment the same as our `.with_cycles` flow that we use elsewhere? If so, should we not use that instead? If not, what is this payment?

**Review by dansteren — CHANGES_REQUESTED** [review]
Everything looks good, other than the change to `cleanDeploy`. But because I have questions on the payment flow I'm requesting changes for now... But let's review and I'm happy to approve this once we talk through the payment stuff.

File: index.ts (hunk 2 above)
Comment (lastmjs) [reply index.ts:129]: It's not the same, it's entirely separate. This is like call_raw, you just call it like a function and pass in a payment if you wish

**Review by bdemann — CHANGES_REQUESTED** [review]
Please add the tests to the test.yaml

**Pull request merged**